src/shared_gui_components/BuildingComponentDialogCentralWidget.cpp

```diff
@@ -169,53 +168,90 @@ void BuildingComponentDialogCentralWidget::setTid() {
   requestComponents(m_filterType, m_tid, m_pageIdx, m_searchString);
 }
 
+std::vector<openstudio::BCLSearchResult> BuildingComponentDialogCentralWidget::fetchAndSortResponses(const std::string& filterType, int tid, const QString& searchString) {
+  m_allResponses.clear();
+
+  RemoteBCL remoteBCL;
+  remoteBCL.setTimeOutSeconds(m_timeoutSeconds);
+
+  std::vector<BCLSearchResult> responses;
+  int totalPages = 1;
+  int currentPage = 0;
+
+  // Collect all responses from all pages
+  do {
+    std::vector<BCLSearchResult> pageResponses;
+    if (filterType == "components") {
+      pageResponses = remoteBCL.searchComponentLibrary(searchString.toStdString(), tid, currentPage);
+    } else if (filterType == "measures") {
+      pageResponses = remoteBCL.searchMeasureLibrary(searchString.toStdString(), tid, currentPage);
+    }
+    responses.insert(responses.end(), pageResponses.begin(), pageResponses.end());
+    totalPages = remoteBCL.numResultPages();
+  } while (++currentPage < totalPages);
+
+  if (!responses.empty()) {
+    std::sort(responses.begin(), responses.end(), [](const BCLSearchResult& a, const BCLSearchResult& b) {
+      return a.name() < b.name();
+    });
+  }
```

Comment:
If I'm following correctly, you just factored a bit of code into a function, and added this bit that sorts the results.

---

Side note: the `if (!responses.empty())` is not technically needed and does not make it faster. https://quick-bench.com/q/5lPbGyQNKctVRq7IdoD_fUVTPt4

(Running it a bunch of times locally produces results where one is sometimes faster than the other, so they're the same).

```
-----------------------------------------------------------
Benchmark                 Time             CPU   Iterations
-----------------------------------------------------------
SortNoIf_Items         4732 ns         4730 ns       147972
SortWithIf_Items       5040 ns         5038 ns       138705
```